### PR TITLE
ec2_key: add a test for the force option

### PR DESCRIPTION
**Pull request by prasadkatti**
##### SUMMARY
ec2_key has an option called 'force'. If you try to create a keypair with the name of an already existing different keypair, the force option lets you control what should happen. When `force=no`, the key should not be overwritten.

This PR adds a test for this case.

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME
`ec2_key`

##### ANSIBLE VERSION

```
devel
```

### CODE REVIEW

File: test/integration/targets/setup_sshkey/tasks/main.yml [hunk 1]
```diff
@@ -32,6 +32,23 @@
   tags:
     - prepare
 
+- name: create another random file
+  shell: mktemp /tmp/id_rsa.XXXXXX
```
Comment (willthames) [inline test/integration/targets/setup_sshkey/tasks/main.yml]: should probably use `tempfile` module for this?

File: test/integration/targets/setup_sshkey/tasks/main.yml [hunk 2]
```diff
@@ -32,6 +32,23 @@
...
+  tags:
+    - prepare
+
+- name: generate another_sshkey
+  shell: echo 'y' | ssh-keygen -P '' -f {{another_sshkey.stdout}}
+  tags:
+    - prepare
+
+- name: record another key_material
+  command: cat {{another_sshkey.stdout}}.pub
```
Comment (willthames) [inline test/integration/targets/setup_sshkey/tasks/main.yml]: use a lookup rather than cat: `lookup('file', '{{ another_sshkey.stdout }}.pub')`

Comment (willthames) [reply test/integration/targets/setup_sshkey/tasks/main.yml]: in fact `another_key_material` should just be set to that lookup.

File: test/integration/targets/ec2_key/tasks/main.yml [hunk 3]
```diff
@@ -147,7 +147,7 @@
            - '"EC2ResponseError: 401 Unauthorized" in result.module_stderr'
 
     # ============================================================
-    - name: test state=absent with key_material
+    - name: test removing a non-existant keypair
```
Comment (willthames) [inline test/integration/targets/ec2_key/tasks/main.yml]: typo: non-existent

**Review by willthames — CHANGES_REQUESTED** [review]

File: test/integration/targets/ec2_key/tasks/main.yml [hunk 4]
```diff
@@ -300,6 +300,26 @@
            # - 'result.key.fingerprint == "{{fingerprint}}"'
 
     # ============================================================
+
+    - name: test force=no with another_key_material (expect changed=false)
+      ec2_key:
+        name='{{ec2_key_name}}'
```
Comment (willthames) [inline test/integration/targets/ec2_key/tasks/main.yml]: We should use full yaml style:

```
  name: '{{ ec2_key_name }}'
  etc...
```

**Comment by prasadkatti** [comment]
ready_for_review

**Review by willthames — APPROVED** [review]

**Comment by willthames** [comment]
Merged, thanks @prasadkatti

**Pull request merged**